### DISCUSSION TITLE
Scoped packages vs pixi.js package when using v7

### DISCUSSION THREAD

**Discussion by Emobe** [pixijs/pixijs#9006] [Q&A] [answered]
When using pixi v7, my assumption is the scoped packages are the way to go but then I saw #8798 and I was wondering if I should just install every package in https://pixijs.io/customize/ until I know exactly what I need and don't need.

Seeing as I use rollup in my projects, would tree shaking still work anyway?

Also if I had a project that did require every package anyway, I assume it is better to use the scoped packages as opposed to importing from `pixi.js`.

I guess my question is with v7, should we be using scoped packages over `pixi.js`?

**Comment by bigtimebuddy** [comment, marked as answer]
> **Warning**
> https://pixijs.io/customize hasn't been updated for v7 yet. That's on my TODO list.

Good question. 

If you are making 3rd-party plugins for PixiJS, I'd strongly suggest using the base packages (e.g., `@pixi/core`) instead of the bundles (`pixi.js`, `pixi.js-legacy` etc).

As for your project, it depends! The advantage to using the base packages is it allows you the smallest possible footprint (if filesize is a major concern). There are no real performance benefits. Because we install a lot of plugins statically, tree-shaking doesn't do _a lot_ so you actually have to exclude the packages for things you don't need.

In general, most users should be using `pixi.js` or `pixi.js-legacy`. I would consider using base packages as an optimization in the final phase of a project.

**Comment by miltoncandelero** [comment]
As a side note, if you try to use scoped packages with v7, keep in mind that now things register by side effects instead of registering manually.
in other words, before you had to register your things from scoped packages yourself, now you have to use a loose `import "@pixi/packagename"` so that the side-effect runs and registers your thingy (applies to a lot of things, mostly assets and renderers)
